[1.68.0] DO NOT MERGE UNTIL RELEASE 45256 Document ability to make a release required (#251)
replicatedhq/replicated-docs: docs/vendor/releases-promoting.md

  The Promote Release dialog opens.

1. Edit the fields:
    * **Channel**: Select the channel where you want to promote the release. The defaults are Stable, Beta, and Unstable. If you created custom channels using the CLI, they are listed here also.

>     * **Version label**: Enter a version label. You can use the semantic version format if you are using the Replicated app manager v1.58.0 or later. If you are using semantic versioning, you must also enable the semantic versioning toggle in the channel settings. This toggle is disabled by default. If you use semantic versioning on a channel, we recommend that you always use semantic versioning on that channel. For more information, see [Enabling Semantic Versioning](releases-semantic-versioning).

Author: Nice, I think this is a big improvement. LMK if there is more here that you think it's worth opening a separate docs story for, but will go ahead and commit this suggestion to this PR.

    * **Requirements**: Select **Prevent this release from being skipped during upgrades** to mark the release as required in the Replicated admin console. When a release is required, the admin console requires users to upgrade to that version before they can upgrade to a later version. For example, if you select **Prevent this release from being skipped during upgrades** for release v2.0, users must upgrade to v2.0 before they can upgrade to a version later than v2.0, such as v2.1.

      :::note

>       When you mark a version as required and `allowRollback` is set to `true` in the Application custom resource manifest, users can no longer re-deploy, or _roll back_ to, versions earlier than the required version. For more information about rollbacks, see [allowRollback](../reference/custom-resource-application#allowrollback) in the _Application_ custom resource topic.

Reviewer: The required version needs to be deployed before the non-rollback scenario occurs. Does this still work?
```suggestion
      Even if `allowRollback` is set to `true` in the Application custom resource, users cannot redeploy, or _roll back_, to versions earlier than the required version once the required version is deployed. For more information about rollbacks, see [allowRollback](../reference/custom-resource-application#allowrollback) in the _Application_ custom resource topic.
```

Author: Nice, good clarification. I'm going to push a new commit to rewrite this suggestion a touch. 

q: saw that you removed the word manifest. We've been using "custom resource manifest file"/"custom resource manifest" when referring to the actual yaml manifest file for a custom resource. Is this incorrect? Want to make sure we update that guidance if so.

Reviewer: That was just an oversight. I put a comma in the wrong place here too. Whoops!

Author: ![Screen Shot 2022-03-25 at 11 00 19 AM](https://user-images.githubusercontent.com/11523028/160167147-1ce43ad9-f03d-4f06-ae48-8a60d97441ed.png)

Here's the updated note-- Very similar to your suggestion, just flipped around the order so that the most important stuff comes first.

Reviewer: Perf

  The Promote Release dialog opens.

1. Edit the fields:
    * **Channel**: Select the channel where you want to promote the release. The defaults are Stable, Beta, and Unstable. If you created custom channels using the CLI, they are listed here also.
    * **Version label**: Enter a version label. If semantic versioning is enabled for the channel, you must use a valid semantic version. For more information, see [Enabling Semantic Versioning](releases-semantic-versioning).
    * **Requirements**: Select **Prevent this release from being skipped during upgrades** to mark the release as required. When a release is required, the admin console requires users to upgrade to that version before they can upgrade to a later version. For example, if you select **Prevent this release from being skipped during upgrades** for release v2.0.0, users with v1.0.0 deployed must upgrade to v2.0.0 before they can upgrade to a version later than v2.0.0, such as v2.1.0.

      :::note

>       After users deploy a required version, they can no longer redeploy, or _roll back_ to, versions earlier than the    required version, even if `allowRollback` is `true` in the Application custom resource manifest. For more information about rollbacks, see [allowRollback](../reference/custom-resource-application#allowrollback) in the _Application_ custom resource topic.

Reviewer: ```suggestion
      After users deploy a required version, they can no longer redeploy, or _roll back_ to, versions earlier than the required version, even if `allowRollback` is `true` in the Application custom resource manifest. For more information about rollbacks, see [allowRollback](../reference/custom-resource-application#allowrollback) in the _Application_ custom resource topic.
```
